src/Photons.ts

```diff
@@ -12,63 +13,53 @@ import Complex, { Cx } from './Complex'
  * x, y, direction, polarization
  * @see {@link @Dimension.position}, {@link @Dimension.direction}, {@link @Dimension.polarization}
  * Designed so that it will work with https://github.com/stared/quantum-game-2 board.
- * @todo Think deeply about which things should change in-plance, and which: modify this object.
- * @todo A lot of things with interfaces to make them consistent.
+ * @todo Think deeply about which things should change in-place, and which things modify this object.
+ * @todo Should compute with numerical values and convert to string for display/debug instead
+ * of using H/V in computation which forbids non 90 values
  */
 export default class Photons {
-  readonly sizeX: number
-  readonly sizeY: number
-  vector: Vector
-  nPhotons: number
   readonly dimX: Dimension
   readonly dimY: Dimension
+  vector: Vector
+  nPhotons: number
 
   /**
    * Create a board for photons.
    * @param sizeX An integer, size x (width) of the board.
    * @param sizeY An integer, size y (height) of the board.
    */
   constructor(sizeX: number, sizeY: number) {
-    this.sizeX = sizeX
-    this.sizeY = sizeY
-    this.vector = new Vector([], [])
-    this.nPhotons = 0
     this.dimX = Dimension.position(sizeX, 'x')
     this.dimY = Dimension.position(sizeY, 'y')
+    this.vector = new Vector([], [])
+    this.nPhotons = 0
   }
 
   /**
    * @returns A deep copy of the same object.
    */
   copy(): Photons {
-    const newPhotons = new Photons(this.sizeX, this.sizeY)
+    const newPhotons = new Photons(this.dimX.size, this.dimY.size)
     newPhotons.vector = this.vector.copy()
     newPhotons.nPhotons = this.nPhotons
     return newPhotons
   }
 
   /**
-   * Normalize in-place
+   * @returns a normalized deep copy
    */
-  normalizeInplace(): void {
-    this.vector = this.vector.normalize()
-  }
-
-  /**
-   * Normalize a copy
-   */
-  normalize(): Photons {
+  normalizedCopy(): Photons {
     const newPhotons = this.copy()
-    this.normalizeInplace()
+    newPhotons.vector.normalize()
     return newPhotons
   }
 
   /**
-   * Dimension indices for particle i, for [posX, posY, dir, pol].
+   * Dimension indices for photon i, for [posX, posY, dir, pol].
    * @param i Photon id, from [0, ..., nPhotons - 1]
    * @returns E.g. for 1: [4, 5, 6, 7]
    */
-  vectorIndicesForParticle(i: number): number[] {
+  vectorIndicesForPhoton(i: number): number[] {
```

Comment:
A lot of name changes, without a good reason. Not a big fan of that if without prior discussion.

Reply:
- The current naming is inconsistent, I think we should use photon instead of Particle since this is the name of the file and it is used pretty much everywhere else.
- The sizeX and sizeY are never used and can be accessed with dimX.size.
- Normalize is unclear and used for a deep copy operation, should be renamed to reflect use.